cardano-cli/src/Cardano/CLI/Legacy/Option.hs

```diff
@@ -378,3 +378,33 @@ pGenesisCmds envCli =
         , Opt.help "Each bulk pool to contain this many pool credential sets [default is 0]."
         , Opt.value 0
         ]
+
+pShelleyBasedEra :: EnvCli -> Parser AnyShelleyBasedEra
+pShelleyBasedEra envCli =
+  asum $
+    [ flag' (AnyShelleyBasedEra ShelleyBasedEraShelley) $
+        mconcat [long "shelley", Opt.help "Specify the Shelley era"]
+    , flag' (AnyShelleyBasedEra ShelleyBasedEraAllegra) $
+        mconcat [long "allegra", Opt.help "Specify the Allegra era"]
+    , flag' (AnyShelleyBasedEra ShelleyBasedEraMary) $
+        mconcat [long "mary", Opt.help "Specify the Mary era"]
+    , flag' (AnyShelleyBasedEra ShelleyBasedEraAlonzo) $
+        mconcat [long "alonzo", Opt.help "Specify the Alonzo era"]
+    , flag' (AnyShelleyBasedEra ShelleyBasedEraBabbage) $
+        mconcat [long "babbage", Opt.help "Specify the Babbage era"]
+    , flag' (AnyShelleyBasedEra ShelleyBasedEraConway) $
+        mconcat [long "conway", Opt.help "Specify the Conway era"]
+    ]
+      ++ (pure <$> maybeToList (envCliShelleyBasedEra envCli))
+
+envCliShelleyBasedEra :: EnvCli -> Maybe AnyShelleyBasedEra
+envCliShelleyBasedEra envCli = do
+  AnyCardanoEra era <- envCliAnyCardanoEra envCli
+  case era of
+    ByronEra -> Nothing
+    ShelleyEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraShelley
+    AllegraEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraAllegra
+    MaryEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraMary
+    AlonzoEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraAlonzo
+    BabbageEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraBabbage
+    ConwayEra -> Just $ AnyShelleyBasedEra ShelleyBasedEraConway
```

Comment:
Let's try to avoid casing on every era wherever it's possible. It'll mean less places to update for the new era.
```suggestion
  AnyShelleyBasedEra $ forEraMaybeEon era
```